main/src/io/arg_parser.hpp

```diff
@@ -89,9 +91,31 @@ bool isExtraOutputStep(size_t step, double t1, double t2, const std::vector<std:
     return std::any_of(extraOutputs.begin(), extraOutputs.end(), matchStepOrTime);
 }
 
-bool isPeriodicOutputStep(size_t step, int writeFrequency)
+/*! @brief Evaluate if the current step should be output (to file) per time frequency
+ *
+ * @param t             simulation time at beginning of current step
+ * @param dt            time step of the step
+ * @param frequency     frequency time to output the simulation (>0.)
+ * @return              true if the step have a time that fit with the usual time frequency output in [t, t+dt]
+ */
+bool isPeriodicOutputTime(double t, double dt, double frequency)
+{
+    assert(frequency > 0.);
+    double currentTime = std::fmod(t, frequency);
+    double futureTime  = currentTime + dt;
```

Comment:
There's a subtle complication here: When this function is called, the current time is `d.ttot` and at the beginning of
the timestep, it was `d.ttot - d.minDt`. These are the two times that we have access to.
`d.ttot + d.minDt` is not guaranteed to be the simulation time of the next time step, because the dt that is added to `d.ttot` will be calculated in the next time step.